samza-core/src/main/scala/org/apache/samza/container/SamzaContainer.scala

```diff
@@ -792,6 +792,10 @@ class SamzaContainer(
     try {
       info("Starting container.")
 
+      if (containerListener != null) {
```

Comment:
I think this is pending the design and fix for SAMZA-1168. When we make StreamProcessor also to be the processor launched in YARN container process and SamzaContainer is completely an internal object inside StreamProcessor, this containerListener interface may either change or not needed anymore.

Reply:
Took a look at the current implementation of both classes and you are right that ContainerListener should be removed. Pending on SAMZA-1168 when we move the SamzaContainer as an internal object to StreamProcessor in YARN container processes, we may introduce a different ContainerListener interface than the ProcessorLifecycleListener later. Deleting the ContainerListener now.

Reply:
Looking at the Javadoc in SamzaContainerListener and its usage in StreamProcessor, the interface itself still make sense and should serve a different purpose than ProcessorLifecycleListener. The problem right now is that SamzaContainer and StreamProcessor are both used as top-level object to run in various environment (i.e. SamzaContainer in ThreadJob/ProcessJob/LocalContainerRunner, and StreamProcessor in LocalApplicationRunner), which is confusing. Since the interface definition itself still make sense and probably will be used even after SAMZA-1168 refactoring, I am leaving it as is now. Fixing ThreadJobFactory to invoke user defined ProcessorLifecycleListener here.